cairo-contracts/packages/clients/src/cometbft/component.cairo

```diff
@@ -470,6 +508,24 @@ pub mod CometClientComponent {
             self.client_sequence.write(client_sequence);
         }
 
+        fn write_update_height(
+            ref self: ComponentState<TContractState>, client_sequence: u64, update_height: Height
+        ) {
+            let mut update_heights = self.update_heights.read(client_sequence);
+
+            if update_heights.len() == 255 {
+                update_heights.pop_front().unwrap();
+            }
+
+            update_heights.append(update_height);
+
+            let mut update_heights_span = update_heights.span();
+
+            let sorted_update_heights = MergeSort::sort(update_heights_span);
```

Comment:
Probably a way to reduce the update cost is to check if the given update height is higher than the latest height, then we can just append and not do any merging.

Reply:
Good point 👌🏻 
[6411a82](https://github.com/informalsystems/ibc-starknet/pull/106/commits/6411a827959194162cae527c76295305663c8d48)